db/seeds.rb

```diff
@@ -68,161 +68,193 @@ def fact_check_attributes(fact_check_link, user, project, team)
-    uid = random_string
-
-    # Tipline user
-    smooch_user_data = {
-    'id': uid,
-    'raw': {
-        '_id': uid,
-        'givenName': tipline_user_name,
-        'surname': tipline_user_surname,
-        'signedUpAt': Time.now.to_s,
-        'properties': {},
-        'conversationStarted': true,
-        'clients': [
-        {
-            'id': random_string,
-            'status': 'active',
-            'externalId': phone,
-            'active': true,
-            'lastSeen': Time.now.to_s,
-            'platform': 'whatsapp',
-            'integrationId': random_string,
-            'displayName': phone,
-            'raw': {
-            'profile': {
-                'name': tipline_user_name
-            },
-            'from': phone
-            }
-        }
-        ],
-        'pendingClients': []
-    },
-    'identifier': random_string,
-    'app_name': random_string
-    }
-
-    fields = {
-    smooch_user_id: uid,
-    smooch_user_app_id: random_string,
-    smooch_user_data: smooch_user_data.to_json
-    }
-
-    Dynamic.create!(annotation_type: 'smooch_user', annotated: team, annotator: BotUser.smooch_user, set_fields: fields.to_json)
-
-    # Tipline request
-    smooch_data = {
-    'role': 'appUser',
-    'source': {
-        'type': 'whatsapp',
-        'id': random_string,
-        'integrationId': random_string,
-        'originalMessageId': random_string,
-        'originalMessageTimestamp': Time.now.to_i
-    },
-    'authorId': uid,
-    'name': tipline_user_name,
-    '_id': random_string,
-    'type': 'text',
-    'received': Time.now.to_f,
-    'text': tipline_text,
-    'language': 'en',
-    'mediaUrl': nil,
-    'mediaSize': 0,
-    'archived': 3,
-    'app_id': random_string
-    }
-
-    fields = {
-    smooch_request_type: 'default_requests',
-    smooch_data: smooch_data.to_json
-    }
-
-    a = Dynamic.new(annotation_type: 'smooch', annotated: project_media, annotator: BotUser.smooch_user)
-    a.set_fields = fields.to_json
-    a.save!
+puts "If you want to create a new user: press 1 then enter"
+puts "If you want to add more data to an existing user: press 2 then enter"
+print ">> "
+answer = STDIN.gets.chomp
+
+ActiveRecord::Base.transaction do
+  if answer == "1"
+    puts 'Making Team / Workspace...'
+    team = create_team(name: Faker::Company.name)
+
+    puts 'Making User...'
+    user = create_user(name: data[:user_name], login: data[:user_name], password: data[:user_password], password_confirmation: data[:user_password], email: Faker::Internet.safe_email(name: data[:user_name]), is_admin: true)
+
+    puts 'Making Project...'
+    project = create_project(title: team.name, team_id: team.id, user: user, description: '')
+
+    puts 'Making Team User...'
+    create_team_user(team: team, user: user, role: 'admin')
+  end
+
+  if answer == "2"
+    puts "Type user name then press enter"
```

Comment:
> even then kept failing for me for some reason (dev data is messy!)

This was because it assumed a created user had a `team`, `project` and `team_user`, which is not the case when we are creating directly from check's landing page.

I'm fixing this by checking if `team_user` is nil and generating everything if needed.

And agree that it's way better to get the user by the email 🙂